vizro-core/src/vizro/models/_layout.py

```diff
@@ -25,6 +26,39 @@ class ColRowGridLines(NamedTuple):
     row_end: int
 
 
+def _place_components_in_grid(grid, components):
+    """Places components inside pre-defined grid."""
+    for idx, component in enumerate(components):
+        grid.children[idx].children = component.build()
+    return grid
+
+
+def _get_unique_grid_component_ids(grid: List[List[int]]):
+    unique_grid_idx = np.unique(grid)
+    unique_grid_comp_idx = unique_grid_idx[unique_grid_idx != EMPTY_SPACE_CONST]
+    return unique_grid_comp_idx
+
+
+# Validators for reuse
+def set_layout(cls, layout, values):
+    from vizro.models import Layout
+
+    if "components" not in values:
+        return layout
+
+    if layout is None:
+        grid = [[i] for i in range(len(values["components"]))]
+        return Layout(grid=grid)
+
+    unique_grid_idx = _get_unique_grid_component_ids(layout.grid)
+    if len(unique_grid_idx) != len(values["components"]):
+        raise ValueError("Number of page and grid components need to be the same.")
+    return layout
+
+
+# TODO: Insert layout only
```

Comment:
Should this TODO still be here?

Reply:
No! Could you remove it for me when you push the other change? :)